lib/dolly/mango_index.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+require 'dolly/document'
+
+module Dolly
+  class MangoIndex
+    class << self
+      DESIGN = '_index'
+
+      def all
+        Dolly::Document.connection.get(DESIGN)[:indexes]
+      end
+
+      def create(name, fields, type = 'json')
+        Dolly::Document.connection.post(DESIGN, build_index_structure(name, fields, type))
```

Comment:
You can use delegate to have `connection.post` method available

`delegate :post, to: Dolly::Document.connection`

Reply:
@javierg  `delegate`  is part of active support and i'd rather not use it so i refactored the class to use `Forwardable`